httplib.h

```diff
@@ -2913,8 +2932,6 @@ class MultipartFormDataParser {
 
   template <typename T, typename U>
   bool parse(const char *buf, size_t n, T content_callback, U header_callback) {
-    static const std::regex re_content_type(R"(^Content-Type:\s*(.*?)\s*$)",
```

Comment:
`^Content-Type:\s*(.*?) \s*$`
The above regex tries to parse string of following type (in order mentioned), 
1. "Content-type:" 
2. list of spaces
3. zero or more characters (could be anything) - matched lazily 
4. List of trailing spaces 

If the part 3 has spaces in between e.g. (application/json   some-string   another-string ) then as it is being lazily matched, it will start considering the spaces in between as part of (4) but then when it encounters any non space character like ('s' from "some-string") it will go back and add it to (3)
This can cause issues if there are lots of spaces in between. And eventually crashes the server. I couldn't think of any simpler regex so added simple string processing instead.